olmo_data/data.py

```diff
@@ -0,0 +1,27 @@
+from contextlib import contextmanager
+from pathlib import Path
+from typing import Iterator
+
+import importlib_resources
+from importlib_resources.abc import Traversable
+
+
+def _get_data_traversable(data_rel_path: str) -> Traversable:
+    return importlib_resources.files("olmo_data").joinpath(data_rel_path)
+
+
+def is_data_dir(data_rel_path: str) -> bool:
+    return _get_data_traversable(data_rel_path).is_dir()
+
+
+def is_data_file(data_rel_path: str) -> bool:
+    return _get_data_traversable(data_rel_path).is_file()
+
+
+@contextmanager
+def get_data_path(data_rel_path: str) -> Iterator[Path]:
```

Comment:
I think the correct return type is:
```suggestion
def get_data_path(data_rel_path: str) -> Generator[Path, None, None]:
```

Reply:
The [python docs](https://docs.python.org/3/library/typing.html#typing.Generator) say

> If your generator will only yield values, set the SendType and ReturnType to None:
> ...
> Alternatively, annotate your generator as having a return type of either Iterable[YieldType] or Iterator[YieldType]:

🤷

Reply:
https://github.com/allenai/OLMo/pull/645/commits/d005b165a2e4c66bda8749bf9e166edbeddf7a08

Reply:
Ok fair enough!

Reply:
I'm with it either way. Your way is less verbose so that's a plus.